src/components/ModalAddComponent/ModalAddComponent.js

```diff
@@ -38,14 +38,20 @@ const ModalAddComponent = ({
       //Filtering system component with system endDate
       const sysWithEndDate = sysComps.filter((sy) => sy.endDate && new Date(sy.endDate) < new Date());
 
+      //Filtering system component with active | no endDate
+      const activeSystem = sysComps.filter((sy) => !sy.endDate || new Date(sy?.endDate) > new Date());
+
       // selecting component from main. that has componentId is equal to sysWithEndDate's componentId
       const componetWithSystemEndDate = main.filter(({ componentId }) => sysWithEndDate.some(({ componentId: sysCompId }) => sysCompId === componentId));
 
+      // remove active components 
+      const componentWithNonActive = componetWithSystemEndDate.filter(({ componentId, componentTypeCode }) => !activeSystem.some(({ componentId: sysCompId, componentTypeCode: sysCompType }) => sysCompId === componentId && componentTypeCode === sysCompType))
```

Comment:
Is it necessary to compare the `componentTypeCode`, or is the `componentId` alone a unique identifier? if both are necessary for uniqueness, then should the comparison be added to line 45 above?

Reply:
`componentTypeCode` is not necessary for uniqueness.
I will remove the `componentTypeCode` from the uniqueness and modify the sorting